controllers/providers/kubernetes/nodes.go

```diff
@@ -32,3 +38,45 @@ func (k *KubernetesClientSet) ListClusterNodes() (*corev1.NodeList, error) {
 	}
 	return nodes, nil
 }
+
+// DrainNode cordons and drains a node.
+func (k *KubernetesClientSet) DrainNode(node *corev1.Node, PostDrainDelaySeconds time.Duration, client kubernetes.Interface) error {
+	if client == nil {
+		return fmt.Errorf("K8sClient not set")
+	}
+
+	if node == nil {
+		return fmt.Errorf("node not set")
+	}
+
+	helper := &drain.Helper{
+		Client:              client,
+		Force:               true,
+		GracePeriodSeconds:  -1,
+		IgnoreAllDaemonSets: true,
+		Out:                 os.Stdout,
+		ErrOut:              os.Stdout,
+		DeleteEmptyDirData:  true,
+		Timeout:             time.Duration(120) * time.Second,
+	}
+
+	if err := drain.RunCordonOrUncordon(helper, node, true); err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
Did you verify this returns an IsNotFound error?

Reply:
So, under apierrors, there are several different type of errors. IsNotFound() returns error when a resource type is not found.
"IsNotFound returns true if the specified error was created by NewNotFound. NewNotFound returns a new error which indicates that the resource of the kind and the name was not found."

However, I wasn't able to catch the error for invalid nodename. I checked the code for RunNodeDrain(),  they do return an error when they can't select pods for eviction, but I didn't see any check for empty list of pods. I mean, when the pod selector returns an empty list, they just move on to the next steps.